### PR TITLE
Added support for float/double atomics (add, sub, min, max)

### PR DESCRIPTION
**Pull request by aneshlya**
Fixes: https://github.com/ispc/ispc/issues/2905
On Xe targets these atomic functions are not supported, so the error is generated.

The tests for the new functionality have been added in the same manner as the existing atomic tests. While the test coverage for atomics is not exhaustive, it seems acceptable because the implementation of atomics is consistent across preprocessor definitions. Similarly, the newly added tests do not cover 100% of the functionality.

### CODE REVIEW

File: docs/ispc.rst [hunk 1]
```diff
@@ -5471,23 +5471,16 @@ There are also ``int64`` equivalents as well as variants that take
...
-  float atomic_max_local(uniform float * uniform ptr, float value)
-  float atomic_swap_local(uniform float * uniform ptr, float value)
-
-For global atomics, only atomic swap is available for these types:
-
-::
-
-  float atomic_swap_global(uniform float * uniform ptr, float value)
-  double atomic_swap_global(uniform double * uniform ptr, double value)
+  float atomic_add_{local,global}(uniform float * uniform ptr, float value)
```
Comment (nurmukhametov) [inline docs/ispc.rst:5479]: I slightly prefer having both explicitly stated. Otherwise, one might miss them when trying to find or grep for them.

**Review by nurmukhametov — APPROVED** [review]
LGTM!

File: docs/ispc.rst (hunk 1 above)
Comment (aneshlya) [reply docs/ispc.rst:5479]: I did it in the same manner as other atomics described https://github.com/ispc/ispc/pull/2907/files#diff-f014868f1ccd5256b5fa04823a766153c82b0eb98c0b3d9a2b4d154e2ee7275aR5464-R5471

**Review by dbabokin — APPROVED** [review]
LGTM.

**Pull request merged**